src/session/src/lib.rs

```diff
@@ -36,18 +39,24 @@ pub struct Session {
     user_info: ArcSwap<UserInfoRef>,
     conn_info: ConnInfo,
     timezone: ArcSwap<Timezone>,
+    configuration_variables: DashMap<SessionConfigOption, SessionConfigValue>,
```

Comment:
my proposal is to avoid using map here for configuration collection. We can create a struct called `ConfigurationVariables` like

```rust
#[derive(Default)]
pub struct ConfigurationVariables {
  postgres_bytea_output: Option<ByteaOutputValue>,
}
```

so that we don't need additional matching for line 152 in handler.rs

Reply:
The main purpose of using`DashMap` is to easily support concurrent update. If using a custom struct we might need something like
```rust
#[derive(Default)]
pub struct ConfigurationVariables {
  postgres_bytea_output: ArcSwap<Option<ByteaOutputValue>>,
}
```
And the update logic might become complex as the supported variables grows.

Reply:
I think it's not a big deal because the session scope variables doesn't update concurrently as it bound to a single TCP connection and `set` statements come in sequence. All we did here is to bypass compiler check. We can even use `ArcSwap` on `ConfigurationVariables` itself. 

Using a map here will become difficult to maintain as it grows because we lost its actual keys.